src/EVM/SMT.hs

```diff
@@ -249,25 +250,35 @@ assertPropsNoSimp config ps =
       toProp :: (Expr EWord, Int) -> Prop
       toProp (e, num) = PEq e (Var (TS.pack ("abst_" ++ (show num))))
 
-    allVars = fmap referencedVars psElim <> fmap referencedVars bufVals <> fmap referencedVars storeVals <> [abstrVars abst]
-    frameCtx = fmap referencedFrameContext psElim <> fmap referencedFrameContext bufVals <> fmap referencedFrameContext storeVals
-    blockCtx = fmap referencedBlockContext psElim <> fmap referencedBlockContext bufVals <> fmap referencedBlockContext storeVals
+    -- Props storing info that need declaration(s)
+    toDeclarePs     = ps <> keccAssump <> keccComp
+    toDeclarePsElim = psElim <> keccAssump <> keccComp
+
+    -- vars, frames, and block contexts in need of declaration
+    allVars = fmap referencedVars toDeclarePsElim <> fmap referencedVars bufVals <> fmap referencedVars storeVals <> [abstrVars abst]
+    frameCtx = fmap referencedFrameContext toDeclarePsElim <> fmap referencedFrameContext bufVals <> fmap referencedFrameContext storeVals
+    blockCtx = fmap referencedBlockContext toDeclarePsElim <> fmap referencedBlockContext bufVals <> fmap referencedBlockContext storeVals
 
     abstrVars :: AbstState -> [Builder]
     abstrVars (AbstState b _) = map ((\v->fromString ("abst_" ++ show v)) . snd) (Map.toList b)
 
+    -- Buf, Storage, etc. declarations needed
     bufVals = Map.elems bufs
     storeVals = Map.elems stores
-    storageReads = Map.unionsWith (<>) $ fmap findStorageReads ps
-    abstractStores = Set.toList $ Set.unions (fmap referencedAbstractStores ps)
-    addresses = Set.toList $ Set.unions (fmap referencedWAddrs ps)
-
-    keccakAssumes
+    storageReads = Map.unionsWith (<>) $ fmap findStorageReads toDeclarePs
+    abstractStores = Set.toList $ Set.unions (fmap referencedAbstractStores toDeclarePs)
+    addresses = Set.toList $ Set.unions (fmap referencedWAddrs toDeclarePs)
+
+    -- Keccak assertions: concrete values, distance between pairs, injectivity, etc.
```

Comment:
```suggestion
    -- Keccak assertions: concrete values, distance between pairs, injectivity, etc.
    --      This will make sure concrete values of Keccak are asserted, if they can be computed (i.e. can be concretized)
```